index.js

```diff
@@ -1,9 +1,33 @@
 'use strict';
-const urlRegex = require('url-regex');
+const urlRegex = require('url-regex')();
 const normalizeUrl = require('normalize-url');
+const queryString = require('query-string');
+const _ = require('underscore');
 
 module.exports = (str, opts) => {
-	const urls = str.match(urlRegex());
-	const ret = urls ? urls.map(url => normalizeUrl(url.trim().replace(/\.+$/, ''), opts)) : [];
-	return new Set(ret);
+	const urls = str.match(urlRegex) || [];
+	const ret = new Set();
+	urls.forEach(url => {
+		_add(url);
+		_extractQueryParams(url);
+	});
+	return ret;
+
+	function _add(url) {
```

Comment:
Don't add functions after `return`. That hurts readability.

Reply:
done